test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -213,6 +253,92 @@ TEST_F(XdsOverrideHostTest, ConnectingSubchannelIsQueued) {
   picker = ExpectState(GRPC_CHANNEL_READY);
   ExpectPickQueued(picker.get(), pick_arg);
 }
+
+TEST_F(XdsOverrideHostTest, DrainingState) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  ASSERT_NE(ExpectStartupWithRoundRobin(kAddresses), nullptr);
+  ApplyUpdateWithHealthStatuses(
+      {{kAddresses[0], XdsHealthStatus::HealthStatus::kUnknown},
+       {kAddresses[1], XdsHealthStatus::HealthStatus::kDraining},
+       {kAddresses[2], XdsHealthStatus::HealthStatus::kHealthy}});
+  auto picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
+  ExpectRoundRobinPicks(picker.get(), {kAddresses[0], kAddresses[2]});
+  ExpectQueueEmpty();
+  // Draining subchannel is returned
+  std::map<UniqueTypeName, std::string> pick_arg{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), pick_arg), kAddresses[1]);
+  ApplyUpdateWithHealthStatuses(
+      {{kAddresses[0], XdsHealthStatus::HealthStatus::kUnknown},
+       {kAddresses[2], XdsHealthStatus::HealthStatus::kHealthy}});
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
+  // Gone!
+  ExpectRoundRobinPicks(picker.get(), {kAddresses[0], kAddresses[2]}, pick_arg);
+}
+
+TEST_F(XdsOverrideHostTest, DrainingNotDrainingGone) {
```

Comment:
I don't understand the purpose of this test.  It looks like the only thing being covered here that isn't covered elsewhere is that we honor the override host when its status is healthy, but that has nothing to do with draining, so the test name and structure are confusing here.

As I mentioned above, I think that we should just have tests for each health status to show that we honor the override host status appropriately.

Reply:
I believe that you are misunderstanding the spec.  As I mentioned elsewhere, the picker needs to look at the health status of the override host, and if that status is not in the configured override host status set, then it needs to ignore the override host.

I think we need to test all of the following cases:

- override host status set contains UNKNOWN, override host status is UNKNOWN => override host will be used
- override host status set does *not* contain UNKNOWN, override host is UNKNOWN => override host will not be used
- override host status set contains HEALTHY, override host status is HEALTHY => override host will be used
- override host status set does *not* contain HEALTHY, override host is HEALTHY => override host will not be used
- override host status set contains DRAINING, override host status is DRAINING => override host will be used
- override host status set does *not* contain DRAINING, override host is DRAINING => override host will not be used (in this case, the host won't exist in the map at all)

Note that all of this is orthogonal to the other tests that cover the special behavior for DRAINING state related to whether the subchannel gets passed to the child and whether we trigger a connection attempt when we pick a DRAINING override host that is in state IDLE.

Once you add the above 6 tests, I don't think we need this `DrainingNotDrainingGone` test at all.